bdfr/downloader.py

```diff
@@ -107,7 +108,22 @@ def _download_submission(self, submission: praw.models.Submission):
             content = downloader.find_resources(self.authenticator)
         except errors.SiteDownloaderError as e:
             logger.error(f"Site {downloader_class.__name__} failed to download submission {submission.id}: {e}")
-            return
+            # Redgifs 401 error / temporary auth token handler
+            if (downloader_class.__name__ == "Redgifs") and (str(e).startswith("Server responded with 401")):
+                print("\n-=-=-=-=-=-=-=-=-=-=-=-\nRedgifs temporary API token out of date")
+
+                # Extracting URL from the error message
+                url_start_index = str(e).rfind("https://")
+                redgifs_url = str(e)[url_start_index:]
+
+                # Getting new temp API token
+                redgifs.Redgifs._get_token(redgifs.Redgifs, redgifs_url)
+
+                # Attempted redownload of link that hit the 401 error. Don't think it works though.
+                redgifs.Redgifs._get_link(redgifs_url)
```

Comment:
Here's the line that attempts to re-run the download with the skipped link from the `401` error.

I'm guessing the chain resolves somewhere else and the link doesn't get passed correctly.
Might be worth moving the entire error checking code into `redgifs.py` and just handle it over there.
Would be easier to pass the url and re-run the download function.